src/sa/sa.c

```diff
@@ -108,6 +108,11 @@ int sa_pton(const char *addr, struct sa *sa)
 	if (inet_pton(AF_INET, addr, &sa->u.in.sin_addr) > 0) {
 		sa->u.in.sin_family = AF_INET;
 	}
+	else if (!strncmp(addr, "unix:", 5)) {
```

Comment:
what if the string is 5 bytes ?

Reply:
Since the provided `addr` string should be always zero terminated, sun_path is empty in this case too.